src/api/stats/getApys.js

```diff
@@ -33,7 +37,20 @@ const updateApys = async () => {
         console.warn('getApys error', result.reason);
         continue;
       }
-      apys = { ...apys, ...result.value };
+
+      // Set default APY values
+      let mappedApyValues = result.value;
+      let mappedApyBreakdownValues = result.value;
+
+      let hasApyBreakdowns = 'apyBreakdowns' in result.value;
+      if (hasApyBreakdowns) {
+        mappedApyValues = result.value.apys;
+        mappedApyBreakdownValues = result.value.apyBreakdowns;
+      }
+
+      apys = { ...apys, ...mappedApyValues };
+
+      apyBreakdowns = { ...apyBreakdowns, ...mappedApyBreakdownValues };
```

Comment:
Do we want to add a default value to the apyBreakdowns? I think this line should be in the `if (hasApyBreakdowns)` statement, that way if the UI doesn't find a breakdown from the breakdown route it can default to the legacy apy endpoint values

Reply:
I hope that’s in line 43, defaults previous to the has command. Does that cover what you meant?

Reply:
So what i mean is we have two final objects, apys and apyBreakdowns. If `apyBreakdowns` is not in result.value, then mappedApyBreakdownValues will look like the same object as the current apy object (= result.value). Then we'll have a mix of formats in the new endpoint, i.e.
```json
{
  "bifi-maxi": 0.13,
  "polycat-pzap-quick": {
        "simpleApy": ".13",
        "compoundingsPerYear": 1000, 
        // ...
      }
  }
}
```

Basically i think the formats shouldn't be mixed for the endpoint. If the breakdown is missing, it should be missing from the new endpoint.

Reply:
Its much easier for the app to check for a missing property (`if (key in  map)`) than to try to differentiate between two types, especially since we're using javascript instead of typescript

Reply:
Right, understand what you’re saying. Thanks for explaining again. 

This was by design, so that logic would be done api side rather than app side. 

It effectively removes the need for ‘if not exists in A, go fetch from B’ so reduces calls. I am comfortable with and desired the hybrid mapping personally. The endpoint represents vaults APY as broken down as possible, which in the case of many is currently not at all.

Are your views on this as strong as blocking until changed?

Reply:
So the apys for all vaults are already located in the same endpoint. We fetch the apys for all vaults periodically in the app, with this one api call. With this new endpoint, we'd be adding the extra fetch call to the periodic fetch, now making 2 calls per interval. So we won't be saving from making extra calls, since the calls aren't individualized to the vaults.

I don't like having mixed formats in an api as it doesn't follow good rest practices. By having it be the same format, its easily described by a OpenAPI/Swagger spec or typescript interface, and makes it easy for others to consume. Think about the interface that entails:

```ts
interface BreakdownApy {
  [poolName: string]: number | { simpleApy: number, compoundingsPerYear: number, /*...*/ }
}
```

Anyone who wants to consume this api will need some custom logic to differentiate between the two formats.

This scenario described in this answer is indeed a contrived. We don't use typescript, and the api is mainly internal to beefy. But it still makes it easier to consume for our app

Reply:
Thanks for going over how the app will consume this. I’m now in full agreement with you that this will be better as just the values that it has been possible to break down the APYs for. Will change and come back. Thanks for leaning into conversation.

Reply:
I think you guys might just be misunderstanding each other?

@Moonster-BSC Sounds like you want the data returned like this?
```json
{
  "venus-wbnb": {
    "totalApy": 0.06983201937728578,
  },
  "boo-bifi-ftm": {
    "simpleApy": 1.1845128790579593,
    "compoundingsPerYear": 2190,
    "beefyPerformanceFee": 0.045,
    "vaultApy": 2.0984988299993503,
    "lpFee": 0.002,
    "tradingApr": 0.052152730701897375,
    "totalApy": 2.2643800119210837
  },
}
```

Instead the current output of:
```json
{
  "venus-wbnb": 0.06983201937728578,
  "boo-bifi-ftm": {
    "simpleApy": 1.1845128790579593,
    "compoundingsPerYear": 2190,
    "beefyPerformanceFee": 0.045,
    "vaultApy": 2.0984988299993503,
    "lpFee": 0.002,
    "tradingApr": 0.052152730701897375,
    "totalApy": 2.2643800119210837
  },
}
```

Reply:
This can work as well. As the "missing breakdown" format is a subset of the breakdown format. In this case the check is similar to a missing check. if (tradingApy in pool).

Reply:
This makes the interface: 
```ts
interface BreakdownApy {
  [poolName: string]: { 
        simpleApy: number,
        tradingApr?: number, // note the nullable ? 
        /*...*/ }
}
```

Reply:
But actually, now totalApy is kinda like how it is right now: we're not sure if it has trading in it or not from the number itself, we have to inspect the rest of the objects properties

Reply:
So i'd rather have the property in the missing breakdown be something like `"farmApy": 0.13, where farmApy is compounded farmAPR (simpleAPY is farmAPR, but honestly should be renamed). Like this:

```ts
{
  "venus-wbnb": {
     "farmApr": 0.05
    "farmApy": 0.06983201937728578,
  },
  "boo-bifi-ftm": {
    "farmApr": 1.12, // don't think this number is necessary tho
    "farmApy": 1.1845128790579593,
    "compoundingsPerYear": 2190,
    "beefyPerformanceFee": 0.045,
    "vaultApy": 2.0984988299993503,
    "lpFee": 0.002,
    "tradingApr": 0.052152730701897375,
    "totalApy": 2.2643800119210837
  },
}
```

Reply:
I hadn’t actually thought of the approach of aligning the schema from your response, so thanks for that. I like this approach and also like your differentiation of the non fee totalApy -> vaultApy and the totalApy when full calculation is available. I will make changes to this now

Reply:
Just to clarify, are you going with @ReflectiveChimp 's suggestion, and what we have in my last comment? I think thats the best of both worlds: we get the functionality you were looking for (don't need to rely on fallback data from another endpoint), and we maintain a clean format

Reply:
`totalApy` is never going to be the real total for **every single vault** unless they **all** have endpoints for trading fees.
So its really "knownTotalApy". 

Given this, probably okay to set `totalApy` = `vaultApy` for legacy data; making `totalApy` required -> i.e. the value consumers of API can rely on to exist and display with the rest being optional breakdown data?

Not a huge deal but otherwise for the 'Overview APY' we'd have to `totalApy in apys ? apys.totalApy : apys.vaultApy`

Reply:
My opinion isn't is strong on totalApy vs vaultApy. I think your argument for going with totalApy is reasonable. I was more concerned about the first issue regarding data format

Reply:
Thanks @ReflectiveChimp for the formatting suggestion

Reply:
Very good, thanks both. For total clarity, that leads us to:

{
  "venus-wbnb": {
    "totalApy": 0.06983201937728578,
  },
  "boo-bifi-ftm": {
    "vaultApr": 1.1845128790579593, // rename from simpleApy
    "compoundingsPerYear": 2190,
    "beefyPerformanceFee": 0.045,
    "vaultApy": 2.0984988299993503,
    "lpFee": 0.002,
    "tradingApr": 0.052152730701897375,
    "totalApy": 2.2643800119210837
  },
}

Just also calling out that without going through other lp files, the vaultApr that appeared in the previous comment won't appear as that's not currently passed through. We will have totalApy in both broken down and non broken down cases and know that some vaults don't have the ability to be broken down yet. Will rename simpleApy as vaultApr as you suggested. Completely agree on the aligned schema with nullable now we've been through this discussion, we should be strong on that.